backend/uclapi/common/decorators.py

```diff
@@ -160,6 +160,16 @@ def _check_oauth_token_issues(token_code, client_secret, required_scopes):
         response.status_code = 400
         return response
 
+    if token.app.deleted:
+        response = JsonResponse({
+            "ok": False,
+            "error":
+                "The token is invalid as the developer has "
+                "deleted their app."
+        })
+        response.status_code = 400
```

Comment:
nit: maybe a 403 error code instead? Or 401 if we are concerned about giving away information that this was a valid token (and change it so that we return 401 if token is invalid/missing).

Reply:
(just realised we give a specific reason so 403 is fine here)

Reply:
good shout, done!